src/function/uvc.c

```diff
@@ -138,6 +139,131 @@ struct {
 		.export = usbg_set_config_node_int,		        \
 	}
 
+int h2b(unsigned char c)
+{
+	/* check if char is a number */
+	if ((c >= '0') && (c <= '9'))
+		return c - '0';
+
+	/* check if char is upper case */
+	if ((c >= 'A') && (c <= 'Z'))
+		return c - 'A' + 10;
+
+	/* check if char is lower case */
+	if ((c >= 'a') && (c <= 'z'))
+		return c - 'a' + 10;
+
+	return -EINVAL;
+}
+
+int usbg_write_guid(const char *path, const char *name,
+		      const char *file, const char *buf)
+{
+	int ret;
+	char guid[16];
+	char *tmp = guid;
+
+	if (strlen(buf) != 32)
+		return USBG_ERROR_INVALID_PARAM;
+
+	/* convert hex representation to binary */
+	for (int i = 0; i < 16; i++) {
+		int hi, lo;
+
+		hi = h2b(*buf);
+		buf++;
+
+		lo = h2b(*buf);
+		buf++;
+
+		if (hi < 0 || lo < 0)
+			return USBG_ERROR_INVALID_PARAM;
+
+		*tmp++ = (hi << 4) | lo;
+	}
+
+	ret = usbg_write_buf(path, name, file, guid, 16);
+	if (ret > 0)
+		ret = 0;
+
+	return ret;
+}
+
+int usbg_read_buf_alloc(const char *path, const char *name,
+			   const char *file, char **dest, int len)
+{
+	char buf[USBG_MAX_FILE_SIZE];
+	char *new_buf = NULL;
+	int ret;
+
+	ret = usbg_read_buf_limited(path, name, file, buf, len);
+	if (ret != len)
+		goto out;
+
+	ret = 0;
+
+	new_buf = malloc(16);
+	if (!new_buf) {
+		ret = USBG_ERROR_NO_MEM;
+		goto out;
+	}
+
+	memcpy(new_buf, buf, 16);
+
+	*dest = new_buf;
+out:
+	return ret;
+}
+
+static inline int usbg_get_guid(const char *path, const char *name,
+			      const char *attr, void *val)
+{
+	return usbg_read_buf_alloc(path, name, attr, (char **)val, 16);
+}
+
+int usbg_set_config_node_guid(config_setting_t *root,
```

Comment:
Move the functions to `src/usbg_common_libconfig.c`. There are already helpers for custom formats like mac addresses. Having a reusable handler for guids doesn't hurt.

Reply:
I thought about that. And yes you are right. I move it.

Reply:
Done